test/test_regexp.py

```diff
@@ -0,0 +1,172 @@
+    ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
+    This key is not known by any other names
+    Are you sure you want to continue connecting (yes/no/[fingerprint])? """,
+
+    'HostKeyPrompt existing': r"""
+    The authenticity of host 'github.com (140.82.121.3)' can't be established.
+    ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
+    This host key is known by the following other names/addresses:
+        x:1: lublub.com
+    Are you sure you want to continue connecting (yes/no/[fingerprint])? """,
+
+    'PassphrasePrompt': r"""
+    Enter passphrase for key '/var/home/lis/.ssh/id_rsa': """,
+
+    'PasswordPrompt': r"""
+    lis@srv's password: """,
+
+    'FIDOPINPrompt': r"""
+    Enter PIN for ED25519-SK key /var/home/lis/.ssh/id_ed25519_sk: """,
+
+    'FIDOUserPresencePrompt': r"""
+    Confirm user presence for key ED25519-SK SHA256:fAxxFFykCijTdrVUUjbbi2TWfCWtOiafhuBhgG7siGg
+    """,
+
+    'AskpassPrompt Change password old prompt': r"""
+    Enter lis@srv's old password: """,
+
+    'AskpassPrompt Change password new prompt': r"""
+    Enter lis@srv's new password: """,
+
+    'AskpassPrompt Change password retype prompt': r"""
+    Retype lis@srv's new password: """,
+
+    'AskpassPrompt Keyboard interaction thing (not encountered)': r"""
+    "(%s@%s) %s""",
+
+    'PKCS11PINPrompt': r"""
+    Enter PIN for '/CN=SSH-key/': """
+}
+
+
+STDERR_MESSAGES = {
+    'ChangedHostKeyError': r"""
+    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    @    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
+    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+    IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
+    Someone could be eavesdropping on you right now (man-in-the-middle attack)!
+    It is also possible that a host key has just been changed.
+    The fingerprint for the ED25519 key sent by the remote host is
+    SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
+    Please contact your system administrator.
+    Add correct host key in x to get rid of this message.
+    Offending ED25519 key in x:1
+    Host key for github.com has changed and you have requested strict checking.
+    Host key verification failed.
+    """,
+
+    'UnknownHostKeyError': r"""
+    No ED25519 host key is known for srv and you have requested strict checking.
+    Host key verification failed.
+    """,
+
+    # 'StrictHostKeyChecking=no (unsupported)': r"""
+    # Password authentication is disabled to avoid man-in-the-middle attacks.
+    # Keyboard-interactive authentication is disabled to avoid man-in-the-middle attacks.
+    # lis@github.com: Permission denied (publickey).
+    # """,
+
+    'AuthenticationError': r"""
+    xyz@srv: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
+    """,
+}
+
+
+@pytest.mark.parametrize('msg_id', ASKPASS_MESSAGES)
+def test_categorize_askpass(msg_id: str) -> None:
```

Comment:
Can you please check the result of an unknown askpass message?

Reply:
Good call.